mintpy/save_geodmod.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+############################################################
+# Program is used for extract accumulated displacement of period
+# Author: Lv Xiaoran                                       #
+# Created: August 2019                                     #
+############################################################
+
+import os
+import re
+import time
+import datetime
+import shutil
+import argparse
+import subprocess
+import numpy as np
+
+import mintpy
+import mintpy.workflow  #dynamic import for modules used by pysarApp workflow
+from mintpy.objects import sensor, RAMP_LIST
+from mintpy.utils import readfile, writefile, utils as ut
+from mintpy.defaults.auto_path import autoPath
+
+######################################################################################
+EXAMPLE = """example:
+  save_geodmod.py $SCRATCHDIR/DarbandikhanSenAT72/PYSAR/timeseries_ECMWF_demErr.h5 20171117_20180603 -b 34.2 35.2 45.0 46.3 -y 0.001 -x 0.001 -t unw -o AT20171117_20180603.unw 
```

Comment:
Thinking about what we need for MimtPy, I think the following two calls should be supported. Do we need a `-t ` or `--type` or `--datatype` option (I don't remember what mintpy uses) to figure out whether the argument is velocity or timeseries? It think it can figure it out. 

```
save_geodmod.py timeseries_ECMWF_demErr.h5 -b 34.2 35.2 45.0 46.3 -y 0.001 -x 0.001 --startDate 20171117 --endDate 20180603 --outdir $MODELDIR

save_geodmod.py velocity.h5 -b 34.2 35.2 45.0 46.3 -y 0.001 -x 0.001 --startDate 20171117 --endDate 20180603 --outdir $MODELDIR [--indir arg]
```
if --starDate and --endDate are not given the defaults are the beginning and end dates (set in argparse)

We switch MimtPy mode on with `--multitrack` option.

For `--multitrack --datasets DarbandikhanSenAT72  --DarbandikhanSenDT27`  it would run the previous save_geodmo.py command in the the `$SCRATCHDIR/DarbandikhanSenAT72` and $SCRATCHDIR/DarbandikhanSenDT27` directories (need a function within the script `run_save_geodmod`).

If  `--datasets` is not given it would search for the datasets in the `$SCRATCHDIR` based on the  *.mimt  filename. If name is `Darbandikhan.mimt` it would look for `$SCRATCHDIR/Darbandikhan*/mintpy` directories and process those.

Note: we follow mintpy convention and chdir into `$SCRATCHDIR/DarbandikhanSenAT72` to execute `run_save_geodmod` in there.

Reply:
Please check how `mintpy/defaults/auto_path.py` is used. We may want to use this when  `--multitrack` is on.

Reply:
@ranneylxr, could you respond to the comments above?

If this is solved, @falkamelung, click on `Resolve conversation` please so that I can go forward and merge this request.

Reply:
I am working on it. Once I finish this part, I will push the modified version.